### DISCUSSION TITLE
Integration of student work GSOC?

### DISCUSSION THREAD

**Discussion by HDInfautre** [joomla/joomla-cms#36243] [Q&A]
HI
https://community.joomla.org/gsoc-2021.html
I read carefully the reports of student work that seem to have progressed well. Is there any work incorporated into joomla 4.1?
What will be the follow-up or integration in the next versions of joomla?

**Comment by online-today** [comment]
The Scheduled Tasks Infrastructure (https://github.com/joomla/joomla-cms/pull/35143) is merged into 4.1
The Cookie Manager, Core Enhancements 1-3 exist as pull requests but seem not to have made it into 4.1 (they are not merged before 4.1 beta 1)

What has become of the following GSoC 21 projects I could not reconstruct immediately:
The Front-end Inline Editing
Media Manager
Accessibility

**Reply by bembelimen** [reply]
- Scheduler => 4.1
- Cookie Manager => not finished yet, needs tests + feedback to get it complete. Would then be an option for 4.2
- Core Enhancement => Final two human tests were missing => Up to the community of they want it in Joomla!
- Frontend Inline Editing => see https://github.com/joomla/joomla-cms/discussions/36243#discussioncomment-1797483
- Media Manager => Missing tests [see here](https://github.com/joomla/joomla-cms/pulls/eyvazahmadzada)
- Accessibility => @softforge is currently sponsoring the further development to get the keyboard short cuts done.

**Reply by PhilETaylor** [reply]
Where is the keyboard shortcuts work being done?

**Reply by Krshivam25** [reply]
> Where is the keyboard shortcuts work being done?

https://github.com/joomla-projects/gsoc21_accessibility/pull/8

**Comment by alikon** [comment]
from https://github.com/joomla/joomla-cms/pulls?q=is%3Apr+is%3Aopen+gsoc

- [4.1] [GSoC 21] Cookie Manager https://github.com/joomla/joomla-cms/pull/35278
- [4.1] [GSoC 21] Media Manager https://github.com/joomla/joomla-cms/pull/35250
- and  #34743, #34764, #35101, #35228, #35250

**Comment by HDInfautre** [comment]
HI
Thank you for these integrations which must make these students happy :-)
Several are interesting. The one I was most interested in to make it easier to update articles for beginners doesn't seem to be on the agenda. 4.2 !? https://github.com/joomla-projects/gsoc21_frontend-inline-editing/issues/10

**Reply by bembelimen** [reply]
In general the feedback on GSoC projects (and the future contribution when they are "free for the public") is very low (beside Brian and Phil). So this projects normaly went a way of abandonware as 80% finished.
This year we invested a lot of energy in certain projects to get them ready and mergeable and others to get them nearly-ready.
Many enhancements were included in 4.0 already, so they were never on the list of future features.

Sadly frontend inline editing had some challenges, so it's far from ready. It can be seen as a very good baseline for future implementations, if someone picks it up or if we do a level 2 GSoC project (therefor mentors are needed...). You see, it's not that easy :D

**Comment by HDInfautre** [comment]
HI
Thank you for taking the time to respond.
As I saw the articles of joomla magazine, I thought that the GSOC works had a good chance to be integrated!

I understand that it is not easy and for me too who would like to get to make one or 2 tests easy!

It seems to me that the 3 https://github.com/joomla/joomla-cms/pull/35250 are integrated?

I downloaded 4.1 beta 1 with the patch tester.
I would like to contribute 
**1/ to the most urgent ones mentioned in this thread**
e.g. https://github.com/joomla/joomla-cms/pull/35250 on the media manager
If I put 35250 in the search field*, I have no result !

**2/ what interests me the most improvement of the media manager;** 
Notably responsive (including the pdf responsive see other post 10 days ago)
How to find those related to the media manager?

**3/ selecting patches**
If I put "filter RTC patches" is that what corresponds to those who are most ready to be integrated? I press the button "test the patch". Normally it does not break anything and it is not more difficult? 
*(I often have a help when I hover over the search field, which hinders me to type. This does not make my task easier)

I will try again before the end of the year. Otherwise not being a developer, I will contribute otherwise.

![FireShot Capture 081 - Joomla! Patch Tester - Test j41 - Administration - joom41test](https://user-images.githubusercontent.com/6817936/145818988-3fc6b1b8-2f02-4834-830a-46eeb7978501.png)


regards

**Reply by dgrammatiko** [reply]
Fwiw the media manager should be done from scratch with sensible approach to the ui/ux. The current implementation should be a huge no as it introduces numerous more input fields when the cms should be getting leaner and easier…
